llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/VectorExt/IR/VectorExtAttrs.td

```diff
@@ -99,5 +99,170 @@ def LayoutAttr : IREEVectorExt_Attr<"Layout",
+    required by a particular intrinsic. For example, some AMDGPU matrix
+    multiplication variants require threads to be distributed
+    like:
+
+    0 1 2 3 4
+    5 6 7 8 9
+    --------- --> Duplicates of shape 2x5 distributed over a batch of size 4x5.
+    0 1 2 3 4
+    5 6 7 8 9
+
+    Duplicates are represented using two attributes:
+      - duplicates_per_batch: Number of duplicates in a batch
+      - duplicate_order: Ordering of duplicates, from outermost to innermost
+
+    Finally, threads are distributed in a single duplicate. The thread
+    distribution is represented by:
+
+      - threads_per_duplicate: Sizes of this level of tiling
+      - thread_order: Ordering of dimensions, from outermost to innermost
+
+    Examples of thread distribution over a 8x4 shape:
+
+    {
+      batches_per_subgroup = [2, 1]
+      duplicates_per_batch = [2, 2]
+      threads_per_duplicate = [2, 2]
 
+      batch_order = [0, 1]
+      duplicate_order = [0, 1]
+      thread_order = [1, 0]
+    }
+
+    Distributed tile:
+
+    {
+      [0 2]|[0 2]      0,1,2,3 --> thread ids
+      [1 3]|[1 3]      
+      ------------     [x z]   --> a single duplicate tile
+      [0 2]|[0 2]      [y w]
+      [1 3]|[1 3]
+    }{
+      [0 2]|[0 2]      { ... } --> a single batch tile
+      [1 3]|[1 3]
+      ------------
+      [0 2]|[0 2]
+      [1 3]|[1 3]
+    }
+
+    So, the thread distribution looks like:
+
+    [0 2 0 2]
+    [1 3 1 3]
+    [0 2 0 2]
+    [1 3 1 3]
+    [0 2 0 2]
+    [1 3 1 3]
+    [0 2 0 2]
+    [1 3 1 3]
+
+    3. Elements per Thread
+
+    The final level of tiling, representing the minimum shape of vector that
+    is treated as an atom.
+
+    The elements are placed contigiously with their shape and ordering
+    determined by:
+      - `elements_per_thread`: Sizes of this level of tiling
+      - `element_order`: Ordering of dimensions, from outermost to innermost
+  }];
+
+  let parameters = (ins
+    ArrayRefParameter<"int64_t", "subgroups_per_workgroup">:$subgroupsPerWorkgroup,
+    ArrayRefParameter<"int64_t", "subgroup_order">:$subgroupOrder,
+
+    ArrayRefParameter<"int64_t", "batches_per_subgroup">:$batchesPerSubgroup,
+    ArrayRefParameter<"int64_t", "batch_order">:$batchOrder,
+
+    ArrayRefParameter<"int64_t", "duplicates_per_batch">:$duplicatesPerBatch,
+    ArrayRefParameter<"int64_t", "duplicate_order">:$duplicateOrder,
+
+    ArrayRefParameter<"int64_t", "threads_per_duplicate">:$threadsPerDuplicate,
+    ArrayRefParameter<"int64_t", "thread_order">:$threadOrder,
+
+    ArrayRefParameter<"int64_t", "elements_per_thread">:$elementsPerThread,
+    ArrayRefParameter<"int64_t", "element_order">:$elementOrder
+  );
+
+  // TODO: add custom parser/printer and builder to elide default value array
+  // refs.
+  let assemblyFormat = [{
+    `<` `subgroups_per_workgroup` `=` `[` $subgroupsPerWorkgroup `]` `,`
+        `batches_per_subgroup`    `=` `[` $batchesPerSubgroup `]` `,`
```

Comment:
We might need to rethink the ordering here at some point. This doesn't allow workgroup contiguous accesses beyond a single subgroup or a single batch because batch unrolling has to happen after tiling to subgroups.